server/src/internalClusterTest/java/org/elasticsearch/snapshots/CloneSnapshotIT.java

```diff
@@ -571,6 +575,40 @@ public void testStartCloneWithSuccessfulShardSnapshotPendingFinalization() throw
         assertEquals(getSnapshot(repoName, cloneName).state(), SnapshotState.SUCCESS);
     }
 
+    public void testStartCloneDuringRunningDelete() throws Exception {
+        final String masterName = internalCluster().startMasterOnlyNode(LARGE_SNAPSHOT_POOL_SETTINGS);
+        internalCluster().startDataOnlyNode();
+        final String repoName = "test-repo";
+        createRepository(repoName, "mock");
+
+        final String indexName = "test-idx";
+        createIndexWithContent(indexName);
+
+        final String sourceSnapshot = "source-snapshot";
+        createFullSnapshot(repoName, sourceSnapshot);
+
+        final List<String> snapshotNames = createNSnapshots(repoName, randomIntBetween(1, 5));
+        blockMasterOnWriteIndexFile(repoName);
+        final ActionFuture<AcknowledgedResponse> deleteFuture = startDeleteSnapshot(repoName, randomFrom(snapshotNames));
+        waitForBlock(masterName, repoName);
+        awaitNDeletionsInProgress(1);
+
+        final ActionFuture<AcknowledgedResponse> cloneFuture = startClone(repoName, sourceSnapshot, "target-snapshot", indexName);
+        logger.info("--> waiting for snapshot clone to be fully initialized");
+        awaitClusterState(state -> {
+            for (SnapshotsInProgress.Entry entry : state.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY).entries()) {
+                if (entry.clones().isEmpty() == false) {
+                    assertEquals(sourceSnapshot, entry.source().getName());
```

Comment:
Maybe we could assert here that the shard snapshots are queued?

Reply:
++ I added that now. Technically this is ensured by assertions we have on the cluster state in `SnapshotsService` but practically this makes the test a lot clearer in its intent :)